### PR TITLE
Add support for Macbook Air, which only supports f32 sample type

### PR DESCRIPTION
**Pull request by simon-frankau**
cpal claims that Macbook Airs only support f32 sample type, so, uh, let's support it!

Prefer f32 on the grounds it's the app's native format, allows for better fidelity when shoveled through a mixer, etc.

I have no idea if you actually want to take PRs, but given I was hacking it up locally I might as well share it.

Nice little project!

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -457,4 +482,4 @@ fn main() {
     stream.play().unwrap();
 
     app.run();
-}
\ No newline at end of file
+}
```
Comment (simon-frankau) [inline src/main.rs:485]: My editor believes in newlines at the end of files, and I just gave up fighting it. Sorry!

File: src/main.rs [hunk 2]
```diff
@@ -291,17 +297,36 @@ impl Application {
         let s = self.audio_sink.lock().unwrap();
         let config = s.config.clone();
         let audio_sink = self.audio_sink.clone();
-        let stream = s.device.build_output_stream(
-            &config.into(),
-            move |data: &mut [i16], info: &cpal::OutputCallbackInfo| {
-                let mut audio_sink = audio_sink.lock().unwrap();
-                audio_sink.fill_sound_buffer(data, info);
+        let stream = match s.config.sample_format() {
```
Comment (simon-frankau) [inline src/main.rs:300]: Dunno if there's a prettier/cleaner way to do this...

File: src/main.rs [hunk 3]
```diff
@@ -235,6 +235,11 @@ impl AudioSink {
         }
     }
 
+    fn get_config(device: &cpal::Device, format: cpal::SampleFormat) -> Option<cpal::SupportedStreamConfigRange> {
```
Comment (simon-frankau) [inline src/main.rs:238]: I guess this function could be nested in ::new. Not sure what works best.

**Comment by simon-frankau** [comment]
(I should probably also mention that I've not tested this on hardware that supports i16 samples. Works for f32.)

**Comment by q3k** [comment]
Sorry, I only noticed this PR now! I blame GitHub notifications.

Let me take a look.

File: src/main.rs (hunk 1 above)
Comment (q3k) [reply src/main.rs:485]: Newlines for the newline gods. It's okay :)

File: src/main.rs (hunk 2 above)
Comment (q3k) [reply src/main.rs:300]: This isn't just good, it's good enough :).

**Review by q3k — APPROVED** [review]

**Comment by simon-frankau** [comment]
> Sorry, I only noticed this PR now! I blame GitHub notifications.

Likewise, only just noticed your response. :)

Cheers!

**Pull request merged**